### ISSUE TITLE
问一下，怎么打开存储的文件？

### ISSUE THREAD

**Issue by helinyu** [Tencent/MMKV#98]
例如: 使用mmkv存储数据，在document下面的mmkv文件夹中的mmkv.default 怎么打开？有没有软件打开？ 类似NSUserDefault那样可以直接打开？

**Comment by lingol** [comment]
No such thing. In fact it’s impossible to decode a mmkv file without knowing which type of each key’s value.

**Comment by lingol** [comment]
> if it is based on protobuf, why cannot decode the mmkv file

@endlessbest Because the type of a key's value is not store inside MMKV. That's why you have to provide value type information when calling MMKV's any decoding method.

**Comment by endlessbest** [comment]
> > if it is based on protobuf, why cannot decode the mmkv file
> 
> @endlessbest Because the type of a key's value is not store inside MMKV. That's why you have to provide value type information when calling MMKV's any decoding method.

well, if mmkv is widely used, there will be a tool for decoding, for debug purpose, may be.  BTW, does mmkv provide key-based expiration feature(the value is no longer available if expired) ?

**Comment by lingol** [comment]
> BTW, does mmkv provide key-based expiration feature(the value is no longer available if expired) ?

No, MMKV doesn't do that. But you can wrap one by appending a last-access-time to each key yourself.
